dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -58,9 +60,8 @@ impl InstrumentMemoryOps {
     pub fn finalize(&self, metadata_file_path: &Path) -> anyhow::Result<()> {
         let mut locs = self.mir_locs.lock().unwrap();
         let mut functions = self.functions.lock().unwrap();
-        let locs: Vec<MirLoc> = locs.drain(..).collect();
-        let functions: HashMap<c2rust_analysis_rt::DefPathHash, String> =
-            functions.drain().collect();
+        let locs = locs.drain(..).collect::<Vec<_>>();
```

Comment:
is this a convention that's preferred?

Reply:
Specifying the generic directly in the call instead of as an annotation also has the added benefit that the expression is now self contained and can be freely moved or used in a larger expression without modification.  You don't get that with the type annotation.